fusillade/clouddirectory.py

```diff
@@ -1183,45 +1192,48 @@ def _set_statement(self, statement: str):
                                  path_name=self._path_name
                              ),
                              policy=dict(
-                                 link_name=self.get_policy_name('IAMPolicy'),
-                                 policy_type="IAMPolicy")
+                                 link_name=self.get_policy_name(policy_type),
+                                 policy_type=policy_type)
                              ))
 
-        self._statement = None
+        self.attached_policies[policy_type] = None
 
     @retry(timeout=1, delay=0.1)
-    def _set_statement_with_retry(self, statement):
+    def _set_policy_with_retry(self, statement, policy_type: str = 'IAMPolicy'):
         """
-        Its possible for self._set_statement to fail with resource not found when the node is first created due to
+        Its possible for self._set_policy to fail with resource not found when the node is first created due to
         race conditions in creating new nodes in cloud directory. Retries give the cloud directory time to finish
         creating node before adding a new policy statement.
-        :param statement:
+        :param policy:
         :return:
         """
-        self._set_statement(statement)
+        if policy_type in self.allowed_policy_types:
+            self._set_policy(statement, policy_type)
 
     def _get_attributes(self, attributes: List[str]):
         """
         retrieve attributes for this from CloudDirectory and sets local private variables.
         """
+        if not attributes:
+            attributes = self._attributes
         try:
             resp = self.cd.get_object_attributes(self.object_ref, self._facet, attributes)
         except cd_client.exceptions.ResourceNotFoundException:
             raise FusilladeHTTPException(status=404, title="Not Found", detail="Resource does not exist.")
         for attr in resp['Attributes']:
             self.__setattr__('_' + attr['Key']['Name'], attr['Value'].popitem()[1])
 
-    def get_attributes(self, attributes: List[str]):
-        attrs = dict()
-        if not attributes:
-            return attrs
+    def get_attributes(self, attributes: List[str]) -> Dict[str, str]:
+        """
+        retrieve the attribute values.
```

Comment:
Nit: Not sure that this docstring is necessary.

Reply:
not fixing at this time

Reply:
but i agree, it's  not needed